File: CemQuintas/src/app/Models/cidadao.ts
import { Sexo } from './sexo';
import { Raca } from './raca';
import { Cidade } from './cidade';
import { Uf } from './uf';

export class Cidadao {
  private _id_cidadao: number;
  private _nome: string;
  private _nome_idx: string;
  private _id_sexo: Sexo;
  private _id_raca: Raca;
  private _id_cidade_naturalidade: Cidade;
  private _idade: number;
  private _complemento_idade: string;
  private _nome_mae: string;
  private _nome_pai: string;
  private _rg: string;
  private _data_rg: Date;
  private _id_uf_rg: Uf;
  private _endereco: string;
  private _telefone: string;
  private _observacoes: string;

  public get IdCidadao(): number {
    return this._id_cidadao;
  }

  public set IdCidadao(idCidadao: number) {
    this._id_cidadao = idCidadao;
  }

  public get Nome(): string {
    return this._nome;
  }

  public set Nome(nome: string) {
    this._nome = nome;
  }

  public get NomeIdx(): string {
    return this._nome_idx;
  }

  public set NomeIdx(nomeIdx: string) {
    this._nome_idx = nomeIdx;
  }

  public get Sexo(): Sexo {
    return this._id_sexo;
  }

  public set Sexo(sexo: Sexo) {
    this._id_sexo = sexo;
  }

  public get Raca(): Raca {
    return this._id_raca;
  }

  public set Raca(raca: Raca) {
    this._id_raca = raca;
  }

  public get Cidade(): Cidade {
    return this._id_cidade_naturalidade;
  }

  public set Cidade(cidade: Cidade) {
    this._id_cidade_naturalidade = cidade;
  }

  public get Idade(): number {
    return this._idade;
  }

  public set Idade(idade: number) {
    this._idade = idade;
  }

  public get ComplementoIdade(): string {
    return this._complemento_idade;
  }

  public set ComplementoIdade(complementoIdade: string) {
    this._complemento_idade = complementoIdade;
  }

  public get NomeMae(): string {
    return this._nome_mae;
  }

  public set NomeMae(nomeMae: string) {
    this._nome_mae = nomeMae;
  }

  public get NomePai(): string {
    return this._nome_pai;
  }

  public set NomePai(nomePai: string) {
    this._nome_pai = nomePai;
  }

  public get Rg(): string {
    return this._rg;
  }

  public set Rg(rg: string) {
    this._rg = rg;
  }

  public get DataRg(): Date {
    if (this._data_rg && !(this._data_rg instanceof Date)) {
      this._data_rg = new Date(this._data_rg);
    }
    return this._data_rg;
  }

  public set DataRg(dataRg: Date) {
    this._data_rg = dataRg;
  }

  public get UfRg(): Uf {
    return this._id_uf_rg;
  }

  public set UfRg(uf: Uf) {
    this._id_uf_rg = uf;
  }

  public get Endereco(): string {
    return this._endereco;
  }

  public set Endereco(endereco: string) {
    this._endereco = endereco;
  }

  public get Telefone(): string {
    return this._telefone;
  }

  public set Telefone(telefone: string) {
    this._telefone = telefone;
  }

  public get Observacoes(): string {
    return this._observacoes;
  }

  public set Observacoes(observacoes: string) {
    this._observacoes = observacoes;
  }

}

File: CemQuintas.OR/Cidadao.cs
using System;
using Regisoft;
using System.Collections.Generic ;

namespace CemQuintas.OR
{
	/// <summary>
	/// Gerado por RSClass - Gerador de Classes
	/// </summary>
	[Serializable]
	public /*sealed*/ class Cidadao
	{
		

		
		#region Private Members		

		private long? _id_cidadao; 
		private string _nome; 
		private string _nome_idx; 
		private Sexo _id_sexo; 
		private Raca _id_raca; 
		private Cidade _id_cidade_naturalidade; 
		private int? _idade; 
		private string _complemento_idade; 
		private string _nome_mae; 
		private string _nome_pai; 
		private string _rg; 
		private DateTime? _data_rg; 
		private Uf _id_uf_rg; 
		private string _endereco; 
		private string _telefone; 
		private string _observacoes; 		
		#endregion

		
		
		#region Constructor

		public Cidadao()
		{
			_id_cidadao = null; 
			_nome = null;
			_nome_idx = null;
			_id_sexo = null; 
			_id_raca = null; 
			_id_cidade_naturalidade = null; 
			_idade = null; 
			_complemento_idade = null;
			_nome_mae = null;
			_nome_pai = null;
			_rg = null;
			_data_rg = null; 
			_id_uf_rg = null; 
			_endereco = null;
			_telefone = null;
			_observacoes = null;
		}
		#endregion // End of Default ( Empty ) Class Constuctor

		
		
		#region Public Properties
			
		public virtual long? IdCidadao
		{
			get
			{ 
				return _id_cidadao;
			}
			set
			{
				_id_cidadao = value;
			}

		}
			



		public virtual string Nome
		{
			get
			{ 
				return _nome;
			}

			set	
			{	
				if( value == null )
					throw new ExceptionRS("Informe 'Nome'");
				
				if(  value.Length > 100)
					throw new ExceptionRS("Valor ultrapassa limite em 'Nome'");					

				_nome = value;
			}
		}
			
		public virtual string NomeIdx
		{
			get
			{ 
				return _nome_idx;
			}

			set	
			{	
				if( value == null )
					throw new ExceptionRS("Informe 'NomeIdx'");
				
				if(  value.Length > 100)
					throw new ExceptionRS("Valor ultrapassa limite em 'NomeIdx'");					

				_nome_idx = value;
			}
		}
			
		public virtual Sexo IdSexo
		{
			get
			{ 
				return _id_sexo;
			}
			set
			{
				_id_sexo = value;
			}

		}
			
		public virtual Raca IdRaca
		{
			get
			{ 
				return _id_raca;
			}
			set
			{
				_id_raca = value;
			}

		}
			
		public virtual Cidade IdCidadeNaturalidade
		{
			get
			{ 
				return _id_cidade_naturalidade;
			}
			set
			{
				_id_cidade_naturalidade = value;
			}

		}
			
		public virtual int? Idade
		{
			get
			{ 
				return _idade;
			}
			set
			{
				_idade = value;
			}

		}
			
		public virtual string ComplementoIdade
		{
			get
			{ 
				return _complemento_idade;
			}

			set	
			{	
				if(  value != null &&  value.Length > 9)
					throw new ExceptionRS("Valor ultrapassa limite em 'ComplementoIdade'");					

				_complemento_idade = value;
			}
		}
			
		public virtual string NomeMae
		{
			get
			{ 
				return _nome_mae;
			}

			set	
			{	
				if(  value != null &&  value.Length > 100)
					throw new ExceptionRS("Valor ultrapassa limite em 'NomeMae'");					

				_nome_mae = value;
			}
		}
			
		public virtual string NomePai
		{
			get
			{ 
				return _nome_pai;
			}

			set	
			{	
				if(  value != null &&  value.Length > 100)
					throw new ExceptionRS("Valor ultrapassa limite em 'NomePai'");					

				_nome_pai = value;
			}
		}
			
		public virtual string Rg
		{
			get
			{ 
				return _rg;
			}

			set	
			{	
				if(  value != null &&  value.Length > 15)
					throw new ExceptionRS("Valor ultrapassa limite em 'Rg'");					

				_rg = value;
			}
		}
			
		public virtual DateTime? DataRg
		{
			get
			{ 
				return _data_rg;
			}
			set
			{
				_data_rg = value;
			}

		}
			
		public virtual Uf IdUfRg
		{
			get
			{ 
				return _id_uf_rg;
			}
			set
			{
				_id_uf_rg = value;
			}

		}
			
		public virtual string Endereco
		{
			get
			{ 
				return _endereco;
			}

			set	
			{	
				if(  value != null &&  value.Length > 100)
					throw new ExceptionRS("Valor ultrapassa limite em 'Endereco'");					

				_endereco = value;
			}
		}
			
		public virtual string Telefone
		{
			get
			{ 
				return _telefone;
			}

			set	
			{	
				if(  value != null &&  value.Length > 15)
					throw new ExceptionRS("Valor ultrapassa limite em 'Telefone'");					

				_telefone = value;
			}
		}
			
		public virtual string Observacoes
		{
			get
			{ 
				return _observacoes;
			}

			set	
			{	
				if(  value != null &&  value.Length > 512)
					throw new ExceptionRS("Valor ultrapassa limite em 'Observacoes'");					

				_observacoes = value;
			}
		}
			
		#endregion 
		
		
		#region Public Functions

		#endregion

		
		
		#region Equals And HashCode Overrides
		/// <summary>
		/// local implementation of Equals based on unique value members
		/// </summary>
		public override bool Equals( object obj )
		{
			if( this == obj ) return true;
			if( ( obj == null ) || ( obj.GetType() != this.GetType() ) ) return false;
			Cidadao castObj = (Cidadao)obj; 
			return ( castObj != null ) &&
				( this._id_cidadao == castObj.IdCidadao );
		}
		
		/// <summary>
		/// local implementation of GetHashCode based on unique value members
		/// </summary>
		public override int GetHashCode()
		{
			
			int hash = 57; 
			hash = 27 * hash * _id_cidadao.GetHashCode();
			return hash; 
		}
		#endregion
		
	}
}
